drivers/serial/uart_native_tty_bottom.c

```diff
@@ -0,0 +1,276 @@
+/**
+ * @brief "Bottom" of native ptty uart driver
+ *
+ * Copyright (c) 2023 Marko Sagadin
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "uart_native_tty_bottom.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include <nsi_tracing.h>
+
+/* ISO C99 requires atleast one argument for "..." in a variadic macro, thats why there are two. */
+#define WARN(msg)          nsi_print_warning(msg)
+#define WARN_ARG(msg, ...) nsi_print_warning(msg, ##__VA_ARGS__)
```

Comment:
I do not really follow why we should not just define
`#define WARN(...) nsi_print_warning(__VA_ARGS__)`
But I may be missing something. If that is the case, you can ignore my comment :)